### ISSUE TITLE
Camera Pose Estimation Performance

### ISSUE THREAD

**Issue by ComputerMath** [zju3dv/DetectorFreeSfM#53]
Hello,

I ran the evaluation on ETH3d with code 
```
 python eval_dataset.py +eth3d_sfm=dfsfm.yaml neuralsfm.NEUSFM_coarse_matcher='loftr_official'
```

for checking camera pose estimation from unknown camera parameters including pose.

But, the result's performance got worse than paper's results. (e.g. acc@1 was 0.53, compared to paper's 0.59)
Also, the scene ```door```'s estimation was 0 in all thresholds, which means that reconstruction failed.

Should there be modified setting on ```eth3d_sfm/dfsfm.yaml```? (such as using other matchers rather than LoFTR).

Also, is there any possible parameters to modify for training/evaluation to increase the performance in ```eth3d_sfm/dfsfm.yaml```?

Thanks in advance :)

**Comment by hxy-123** [comment]
Hi, thanks for your feedback. We have re-run multiple times on different machines before the code release. The SfM metric using `loftr_official` on the scene `door` after refinement should be like this:
<img width="288" alt="image" src="https://github.com/zju3dv/DetectorFreeSfM/assets/54060345/2344d0e4-75bf-4547-98f5-e2536414f8c8">. It's weird that the results are all 0 in your evaluation. Please make sure the installed versions of Python, PyTorch, CUDA as well as the COLMAP are identical to our requirements to ensure reproduction. By the way, could you please specify the GPU type you are using? We are also further checking the results on different devices.

**Comment by ComputerMath** [comment]
Thanks for your experiments result. 
I'll recheck it with your specified version...
COLMAP might not be issue because I did use your colmap's github. ```git clone https://github.com/hxy-123/colmap.git```

I have runned the experiments on NVIDIA GeForce RTX 4090 and 3090 on both, so I don't think this is not the issue of the GPU device.
I have run the experiments with ```python 3.9.1``` and ```'torch==2.1.1+cu118'``` 
Maybe I can modify torch version into yours (1.8).

**Comment by hxy-123** [comment]
I think the different torch versions may affect the final results but not that much (`door` auc@1 from 86 -> 0). I think the problem may be [here](https://github.com/zju3dv/DetectorFreeSfM/blob/4a370f186ece48ed73c441283b54caf2f9aeec08/hydra_configs/eth3d_sfm/dfsfm.yaml#L96). In the previous version of config, the `colmap_config.min_model_size` is `3`, which is copied from IMC config during our code clean process. This is different from the original config in our code base. I think this may lead to the failure of scene `door`, as COLMAP will terminate as long as the number of the registered images is larger than `3`. However, setting this parameter larger will force COLMAP to try other initializations for better results.

**Comment by ComputerMath** [comment]
Thanks for your comment :)
I agree that torch version would not that change much.

I'll re-download your code because my version may be old version.

By the way, matcherformer's door result is not acc@1 0.. (it is higher than 80)
so, I think loftr may not work as yours. 

I'll change the parameter of yaml and let you know :)

**Comment by ComputerMath** [comment]
*******************
aucs_2
auc@1: 0.8636072607937592
auc@3: 0.9503059930423321
auc@5: 0.9701835958253993
auc@10: 0.9850917979126997
auc@20: 0.9925458989563498
*******************

okay, changing the parameters got work! (min_model_size)
Thanks for your advice :)

For the sake of solving other potential problem, I'll re-download the code.
